src/components/form/HdRange.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div class='range'>
+    <input type="range"
+      :name="name"
+      :v-model="value"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="updateRangeDecoration"
+    >
+    <div class="range__thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      refault: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+  },
+  watch: {
+    value() {
+      this.input.value = this.value;
+      this.updateRangeDecoration();
```

Comment:
the method `updateRangeDecoration` is already set as callback for @input and @change, wouldn't it get triggered x3 on each change?

Reply:
I personally would keep it only here at the watcher, and leave the events for callbacks passed from a parent component

Reply:
I had to set it on both change and input because IE11 emits change instead of input during the drag. So I have to keep them both. Unfortunately, I can not make it work at the watcher!